src/mono/System.Private.CoreLib/src/System/Reflection/RuntimeAssembly.cs

```diff
@@ -252,7 +252,7 @@ private static void AddPublicNestedTypes(Type type, List<Type> types, List<Excep
 
         public override AssemblyName GetName(bool copiedName)
         {
-            return AssemblyName.Create(_mono_assembly, CodeBase);
+            return AssemblyName.Create(_mono_assembly, get_code_base (this, false));
```

Comment:
Is this an actual fix? What does it return for single-file like targets?

Reply:
In Mono we have two names associated with an assembly.  It's "filename" which corresponds to `Assembly.Location` which is NULL for single-file targets like WebAssembly.  And "name" which is derived from the name that the assembly was called in the bundle which is returned for `Assembly.CodeBase`.

So yea this is the actual fix - the issue is that the CodeBase public getter is tagged as not available in single-file.  But `get_code_base` is not.  And it returns a reasonable value.  So we use it directly.

Reply:
Yeah and I don't think that's correct. The value is exposed via https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Reflection/AssemblyName.cs#L63 which says that the values should be empty string (null value effectively)

Reply:
Thanks.  Switched to using `MonoImage:filename` which will be null for bundles